helpers/framed.js

```diff
@@ -0,0 +1,31 @@
+let framed;
+
+export async function isFramed() {
+	if (framed !== undefined) return framed;
+
+	if ((window.D2L && window.D2L.IsNotAnIframedApp) || window === window.parent) {
```

Comment:
Double-check that accessing `window.parent` cross-origin doesn't throw an exception here. I've needed `try`-`catch` blocks around similar code in the past.

Reply:
This didn't throw when I tested it locally and the docs suggest it shouldn't anymore. For extra safety's sake, however, I can split this checkout and try-catch it anyways. Admittedly I didn't test across our full range of supported browsers so there for sure could be a case where this blows up.